src/IPblackList.cpp

```diff
@@ -0,0 +1,62 @@
+#include "HashOverrideDestIp.hpp"
+#include"TestsConfig.h"
+#include"SFML/Network.hpp"
+//Checks how much requests there sent to the destination Ip from different source Ips, if its the first time this ip is accessed, send the Ip address to Ip blacklist api to verify it's a legitimate site Ip, return true if it's not legitimate and false if it is.
+bool request_density(const std::string& srcIP, const std::string& dstIP, const std::string& rawdata) {
+	// store the data in map as [destination ip, set of source ip's that accessed him]
+	static std::unordered_map<destinationIp, std::set<std::string>, my_string_hash> map;
+	destinationIp* key = new destinationIp(dstIP);
+	if (map.find(*key) == map.end())
+	{
+		std::set<std::string> srcset;

```

Comment:
Not efficient at all. You have too much copies. Just insert directly

Reply:
what Not efficient?
if this destination Ip doesn't exist you create a set with the source Ip and save it in the map.
how would you suggest to do it better?

Reply:
it shoudnt be here see next comment